### PR TITLE
Adding tag option to action

### LINKED ISSUE #1147: Add tags to deployed commit
### Discussed in https://github.com/JamesIves/github-pages-deploy-action/discussions/1146

<div type='discussions-op-text'>

<sup>Originally posted by **germa89** June 17, 2022</sup>
In certain cases and uses, this can be helpful to track documentation versions.

I'm happy to take a look into this.</div>

I think the implementation should not be super difficult, since it is an extra step in the code.

### PR DESCRIPTION
**Pull request by germa89**
## Description

I found out that in some cases, pushing to the doc repo using a tag is convenient.

## Testing Instructions

In theory, it should work by providing the key option `tag`.

```
- name: Deploy 🚀
  uses: JamesIves/github-pages-deploy-action@v4.3.3
  with:
    branch: gh-pages # The branch the action should deploy to.
    folder: build # The folder the action should deploy.
    tag: 'v0.1'
```

## Additional Notes

* This is my first time using TypeScript, so watch out for the errors.

* This PR closes #1147

### CODE REVIEW

**Review by mgalloue — APPROVED** [review]

**Comment by JamesIves** [comment]
Thanks for the contribution. Do you mind creating an issue for this along with rationale so it can be tracked?

**Review by nmalsang — APPROVED** [review]

**Comment by germa89** [comment]
> Thanks for the contribution. Do you mind creating an issue for this along with rationale so it can be tracked?

Hi @JamesIves , done!

File: src/constants.ts [hunk 1]
```diff
@@ -138,7 +149,9 @@ export const action: ActionInterface = {
     ? true
     : getInput('ssh-key'),
   targetFolder: getInput('target-folder'),
-  workspace: process.env.GITHUB_WORKSPACE || ''
+  workspace: process.env.GITHUB_WORKSPACE || '',
+  add_tag: !isNullOrUndefined(getInput('tag')),
```
Comment (JamesIves) [inline src/constants.ts]: Is `add_tag` and `externalRepositoryTarget` necessary? Are we not able to check the existence of `tag` and `repositoryName` instead directly?

File: README.md [hunk 2]
```diff
@@ -172,6 +173,7 @@ By default, the action does not need any token configuration and uses the provid
...
 | `workspace`        | This should point to where your project lives on the virtual machine. The GitHub Actions environment will set this for you. It is only necessary to set this variable if you're using the node module.                                                                                                                                                                      | `with` | **No**   |
+| `tag`              | Add a tag to the commit. Only works when `repository-name` is specified and `dry-run` is not used.                                                                                                                                                                                                                                                                          | `with` | **No**   |
```
Comment (JamesIves) [inline README.md]: Is there any technical limitation as to why this only works when an external repository is used?

File: src/constants.ts (hunk 1 above)
Comment (germa89) [reply src/constants.ts]: Hi @JamesIves , I do not know how to do that. Ideally yes, only checking the existence of those should be sufficient.

I'm happy to implement this change if some guidance is provided.

File: README.md (hunk 2 above)
Comment (germa89) [reply README.md]: There is no technical limitation. I guess I didn't consider local repositories.

I can remove that restriction if you wish.

Comment (JamesIves) [reply README.md]: I think that'd be a good improvement, I'd prefer to not restrict settings to certain conditions

File: src/constants.ts (hunk 1 above)
Comment (JamesIves) [reply src/constants.ts]: I think we can just check to see if these fields have content or not, we do this in a few places by just validating the existence of content; see: https://github.com/JamesIves/github-pages-deploy-action/blob/dev/src/git.ts#L152 for example

**Comment by germa89** [comment]
@JamesIves I have implemented the changes. Let me know if there is anything more I can do.

**Comment by JamesIves** [comment]
Will get to this in the next couple of days, thanks for your patience!

**Comment by JamesIves** [comment]
Looks good to me! Thanks for making the changes

**Pull request merged**